### ISSUE TITLE
[5.3] Eager loaded relationships with string primary key require $incrementing to be set to false

### ISSUE THREAD

**Issue by ctoma** [laravel/framework#15295]
- Laravel Version: 5.3.6
- PHP Version: PHP 7.0.8-2
- Database Driver & Version: 5.7.13

### Description:
After upgrading from 5.1 to 5.3 I noticed one of my relations was no longer working when eager loading. 
```
$test = Booking::where('booking.id', '=', $this->booking_id)
            ->first();

echo $test->channel
//works

echo $test->channel()->get()
//works

$test = Booking::where('booking.id', '=', $this->booking_id)
        ->with('channel')    
    ->first();

echo $test->channel
//null
```
Obviously the relation is setup correctly as the first and second echo works, but the third one is null. After investigating it turns out I didn't have a $incrementing field set, so it was defaulting to true. Laravel then assumes my channel primary key is an int and eager loading always returns null. This is a change in behaviour from 5.1 (or possibly 5.2) where $incrementing did not matter.

Setting $incrementing to false doesn't change the SQL query itself, so it seems to be an issue with how laravel is processing it once it is retrieved from the database. With $incrementing set to true laravel is successfully retrieving the rows from the database (tested by running the SQL manually) but ends up returning null.

### Steps To Reproduce:
Setup a ->belongsTo() relationship on a model that has a varchar primary key. On the related model do not set $incrementing or set it to true.

### Solution:
Personally I would classify this as a bug, but if not then I believe this should at least be mentioned somewhere in the upgrade docs **and** on the Eloquent Relationship docs. It doesn't seem obvious that $incrementing would affect eager-loaded relationships.

**Comment by GrahamCampbell** [comment]
Yeh, that's correct. This is documented on 5.2+.

**Comment by ctoma** [comment]
Okay, what about the second part of what I wrote?
>should be mentioned in the Eloquent Relationship docs. It doesn't seem obvious that $incrementing would affect eager-loaded relationships.

Should I submit a PR there on the docs? I've found a bunch of google results where people are pulling their hair out and couldn't find a solution. As I said before I don't think it's very obvious to anyone that $incrementing would affect specifically eager loaded relationships, especially when the non-eager loaded relationship works fine.
